custom/opm/reports.py

```diff
@@ -416,31 +380,46 @@ class SharedDataProvider(object):
     Data provider for report data that can be shared across rows in an instance
     of a report.
     """
+    vhnd_form_props = [
+        'attend_ANM',
+        'attend_ASHA',
+        'attend_cmg',
+        'big_weight_machine_avail',
+        'child_weight_machine_avail',
+        'func_bigweighmach',
+        'func_childweighmach',
+        'stock_ifatab',
+        'stock_measlesvacc',
+        'stock_ors',
+        'stock_zntab',
+    ]
+
+    def get_all_vhnd_forms(self):
+        key = make_form_couch_key(DOMAIN, xmlns=VHND_XMLNS)
+        return get_db().view(
+            'reports_forms/all_forms',
+            startkey=key,
+            endkey=key+[{}],
+            reduce=False,
+            include_docs=True
+        ).all()
 
     @property
     @memoized
     def _service_dates(self):
```

Comment:
Do a couch lookup to get all VHND forms, then assemble that same data structure based on the forms.